src/protocolgame.cpp

```diff
@@ -3426,15 +3428,16 @@ void ProtocolGame::sendSessionEnd(SessionEndTypes_t reason)
 }
 
 ////////////// Add common messages
-void ProtocolGame::AddCreature(NetworkMessage& msg, const Creature* creature, bool known, uint32_t remove)
+void ProtocolGame::AddCreature(NetworkMessage& msg, std::shared_ptr<const Creature> creature, bool known,
+                               uint32_t remove)
 {
 	CreatureType_t creatureType = creature->getType();
-	const Player* otherPlayer = creature->getPlayer();
-	const Player* masterPlayer = nullptr;
+	auto otherPlayer = creature->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
	const auto otherPlayer = creature->getPlayer();
```